kerchunk/xarray_backend.py

```diff
@@ -0,0 +1,31 @@
+from xarray.backends import BackendEntrypoint
+import xarray as xr
+import os
+from fsspec.implementations.reference import ReferenceFileSystem
+
+
+class KerchunkBackend(BackendEntrypoint):
+    def open_dataset(
+        self,
+        filename_or_obj,
+        **kwargs,
+    ):
+        return my_open_dataset(filename_or_obj, **kwargs)
+
+    open_dataset_parameters = ["filename_or_obj"]
+
+    def guess_can_open(self, filename_or_obj):
+        try:
+            _, ext = os.path.splitext(filename_or_obj)
+        except TypeError:
+            return False
+        return ext in {".json", ".parquet"}
+
+    description = "Open Kerchunk References with Xarray"
+    url = "https://fsspec.github.io/kerchunk/"
+
+
+def my_open_dataset(filename_or_obj, **kwargs):
```

Comment:
Probably want this to have a meaningful name.

And I suspect we'll want to forward keyword arguments to two places, so we'll need `reference_fs_options` (or `storage_options`?) and `open_dataset_options` instead of `**kwargs`.

Reply:
I'd probably have a signature of
```python
def open_reference_dataset(filename_or_obj, storage_options, open_dataset_kwargs):
    pass
```
(note, no `**kwargs`, since those will be caught by the backend entrypoint's `open_dataset`)

If you go with `storage_options`, you might have to disallow passing `fo` in there. If not, you could make `target_protocol`, `target_options`, `remote_protocol` and `remote_options` parameters (or combine them somehow into `target_options` and `remote_options`, not sure). Whatever you decide, `open_dataset_parameters` should list all the options for the backend.